File: src/main/java/com/llacerximo/booklist/domain/service/AuthorService.java
package com.llacerximo.booklist.domain.service;

import com.llacerximo.booklist.common.dto.AuthorDTO;

import java.util.List;

public interface AuthorService {

    List<AuthorDTO> findAll();
    List<AuthorDTO> findAllByName(String search);
    AuthorDTO findById(Long id);
    AuthorDTO insert(AuthorDTO authorDTO);
    AuthorDTO update(AuthorDTO authorDTO);
    void delete(Long id);

}

File: web/src/app/services/author.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpParams} from "@angular/common/http";
import {URL_BASE} from "../shared/utils";
import {Observable} from "rxjs";
import {Author} from "../components/authors/author";

@Injectable({
  providedIn: 'root'
})
export class AuthorService {

	url: string = `${URL_BASE}/authors`

	constructor(private http: HttpClient) { }

	findAll(name?: string): Observable<Author[]> {
		const options = name ? {params: new HttpParams().set('name', name)} : {};
		return this.http.get<Author[]>(this.url, options);
	}

	findById(id: number): Observable<Author> {
		return this.http.get<Author>(`${this.url}/${id}`);
	}

	insert(author: Author): Observable<Author> {
		return this.http.post<Author>(`${this.url}`, author);
	}

	update(author: Author): Observable<Author> {
		return this.http.put<Author>(`${this.url}/${author.id}`, author);
	}

	delete(id: number) {
		return this.http.delete(`${this.url}/${id}`);
	}

}
